src/proof/retainer.rs

```diff
@@ -22,12 +89,31 @@ impl FromIterator<Nibbles> for ProofRetainer {
 impl ProofRetainer {
     /// Create new retainer with target nibbles.
     pub fn new(targets: Vec<Nibbles>) -> Self {
-        Self { targets, proof_nodes: Default::default() }
+        trace!(target: "trie::proof_retainer", ?targets, "Initializing");
```

Comment:
this seems like it would be noisy even for trace level, i think we initialize many proof retainers in practice?